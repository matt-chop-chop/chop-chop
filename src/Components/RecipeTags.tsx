import { Flex, Tag } from "@chakra-ui/react";

type RecipeTagsProps = {
  tags: string[];
};

const RecipeTags = ({ tags }: RecipeTagsProps) => {
  if (tags.length === 0) return null;

  return (
    <Flex flexWrap="wrap" mt={1}>
      {tags.map((tag: string, index: number) => {
        if (tag.length > 0) {
          return <Tag key={index}>{tag}</Tag>;
        }
        return null;
      })}
    </Flex>
  );
};

export default RecipeTags;
