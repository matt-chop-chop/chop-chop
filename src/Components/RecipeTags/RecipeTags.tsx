import { Flex, Tag } from "@chakra-ui/react";
import React from "react";

export type RecipeTagsProps = {
  tags: string[];
};

const RecipeTags = ({ tags }: RecipeTagsProps) => {
  if (tags.length === 0) return null;

  return (
    <Flex flexWrap="wrap" mt={1}>
      {tags.map((tag: string, index: number) => {
        if (tag.length > 0) {
          return <Tag key={index}>{tag}</Tag>;
        }
        return null;
      })}
    </Flex>
  );
};

export default RecipeTags;
