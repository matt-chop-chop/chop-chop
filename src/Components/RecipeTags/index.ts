export { default as RecipeTags } from "./RecipeTags";
