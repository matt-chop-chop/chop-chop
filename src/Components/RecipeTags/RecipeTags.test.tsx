import { shallow } from "enzyme";
import React from "react";
import RecipeTags, { RecipeTagsProps } from "./RecipeTags";

const props: RecipeTagsProps = {
  tags: ["Test Tag 1", "Test Tag 2"],
};

describe("RecipeTags", () => {
  it("is exported from RecipeTags", () => {
    expect(RecipeTags).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<RecipeTags {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("does not render empty tags ", () => {
    const wrapper = shallow(<RecipeTags tags={[props.tags[0], ""]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("returns null if the list of tags is empty ", () => {
    const wrapper = shallow(<RecipeTags tags={[]} />);

    expect(wrapper.type()).toBeNull();
  });
});
