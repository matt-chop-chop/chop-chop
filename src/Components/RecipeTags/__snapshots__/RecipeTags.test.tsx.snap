// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeTags does not render empty tags  1`] = `
<Flex
  flexWrap="wrap"
  mt={1}
>
  <Tag
    key="0"
  >
    Test Tag 1
  </Tag>
</Flex>
`;

exports[`RecipeTags renders as expected  1`] = `
<Flex
  flexWrap="wrap"
  mt={1}
>
  <Tag
    key="0"
  >
    Test Tag 1
  </Tag>
  <Tag
    key="1"
  >
    Test Tag 2
  </Tag>
</Flex>
`;
